Makefile

```diff
@@ -151,13 +166,17 @@ build/$(CONVERTER_JAR): $(CONVERTER_SOURCES) $(RESOURCES)
 %.class: %.java
 	$(JAVAC) -source 7 -target 7 -Xlint:-options -g:none $^
 
-test: all
-	test/smoke-test.sh
-	test/thread-smoke-test.sh
-	test/alloc-smoke-test.sh
-	test/load-library-test.sh
-	test/fdtransfer-smoke-test.sh
-	echo "All tests passed"
+build-test: all build/$(TEST_JAR)
+	echo "Successfully built all tests $(LIB_PROFILER)"
+
+test: all build/$(TEST_JAR)
+	echo "Running tests against $(LIB_PROFILER)"
+	$(JAVA) $(DFlags) -ea -cp "build/test.jar:build/jar/*:build/lib/*" one.profiler.test.Runner $(TESTS)
+
+build/$(TEST_JAR): $(TEST_SOURCES) build/$(CONVERTER_JAR)
+	mkdir -p build/test
+	$(JAVAC) --release 8 -cp "build/lib/*:build/jar/*:build/converter/*" -d build/test $(TEST_SOURCES)
```

Comment:
`build/lib` should not be needed for compiling tests

Reply:
Addressed will reflect in next commit